modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py

```diff
@@ -944,7 +968,46 @@ def copy(self):
         -------
         OpExpr
         """
-        return OpExpr(self.op, self.operands.copy(), self._dtype)
+        op = OpExpr(self.op, self.operands.copy(), self._dtype)
+        if pk := getattr(self, "partition_keys", None):
+            op.partition_keys = pk
+            op.is_rows = self.is_rows
+            op.order_keys = self.order_keys
+            op.lower_bound = self.lower_bound
+            op.upper_bound = self.upper_bound
+        return op
+
+    @_inherit_docstrings(BaseExpr.nested_expressions)
+    def nested_expressions(
```

Comment:
In this function, there are 3 generators for three different entities at once. Please explain how it works and why it is done this way.

Reply:
The generator is used not only for iterating over the nested expressions but also it allows to replace some expressions. The expressions are replaced not in the current object, but a copy is created. This functionality is used by `BaseExpr.translate_input()`, `BaseExpr.fold()` and `_maybe_copy_and_translate_expr()`.